sources/packages/api/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable, Inject } from "@nestjs/common";
+import { Brackets, Connection } from "typeorm";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { MSFAANumber, Student } from "../../database/entities";
+import * as dayjs from "dayjs";
+import { MAX_MFSAA_VALID_DAYS } from "../../utilities";
+import { SequenceControlService } from "../sequence-control/sequence-control.service";
+
+/**
+ * Service layer for MSFAA (Master Student Financial Aid Agreement)
+ * numbers generated for a student.
+ */
+@Injectable()
+export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
+  constructor(
+    @Inject("Connection") connection: Connection,
+    private readonly sequenceService: SequenceControlService,
+  ) {
+    super(connection.getRepository(MSFAANumber));
+  }
+
+  /**
+   * Creates a new MSFAA record with a new number for the specified student.
+   * @param studentId student to have a new MSFAA record created.
+   * @returns Created MSFAA record.
+   */
+  async createMSFAANumber(studentId: number): Promise<MSFAANumber> {
+    const newMSFAANumber = new MSFAANumber();
+    newMSFAANumber.msfaaNumber = await this.consumeNextSequence();
+    newMSFAANumber.student = { id: studentId } as Student;
+    return this.repo.save(newMSFAANumber);
+  }
+
+  /**
+   * Generates the next number for an MSFAA.
+   * @returns number to be used for the next MSFAA.
+   */
+  private async consumeNextSequence(): Promise<string> {
+    let nextNumber: number;
+    await this.sequenceService.consumeNextSequence(
+      "MSFAA_STUDENT_NUMBER",
+      async (nextSequenceNumber: number) => {
+        nextNumber = nextSequenceNumber;
+      },
+    );
+    return nextNumber.toString();
+  }
+
+  /**
+   * Gets an MSFAA record that should be considered as valid.
```

Comment:
Please change to 'a MSFAA record'

Reply:
Please see the comments above, "an MSFAA" seems to be the correct way.